services/prefect/flows/generated_flows.py

```diff
@@ -73,72 +74,54 @@ def initialize_sentry(sentry_dsn: str):
     sentry_sdk.set_tag("flow", prefect.context.flow_name)
 
 
+@task
+def skip_cached_flow(flow_name):
+    logger = prefect.context.get("logger")
+    logger.info(f"No new data. Skipping {flow_name}")
+
+
+@task
+def check_if_skip_cached_flow(cls: ETagCacheMixin):
+    return not cls().check_if_new_data()  # if there is new data don't skip this flow.
+
+
 def create_flow_for_scraper(ix: int, cls: Type[DatasetBase], schedule=True):
```

Comment:
Modifies `create_flow_for_scraper` to check if the class has etag checking, and if it does and there is no new data we skip the scraper. 

Classes without the `EtagMixinClass` will always run